### ISSUE TITLE
Issue while trying to use sr_lock from both netconf client and application

### ISSUE THREAD

**Issue by vshaker** [CESNET/netopeer2#1057]
Hi , 

From both application and netconf client we are trying to access/update the running DS , Netconf side lock is acquired and I tried using sr_lock in the application side and it is coming out with the below error 

**Application will set "/examples:cont/l" to "vimala".
sr_set_item_example|ERROR "SR stderr" Module "yang" is DS-locked by session 75.**

How do we handle this situation of acquiring lock serially . 
 

Verrsion used :
Netopeer2 CLI 2.0.62
Compile time: Sep  7 2021, 13:28:05

sysrepoctl - sysrepo YANG schema manipulation tool, compiled with libsysrepo v1.4.146 (SO v5.6.62)

**Comment by michalvasko** [comment]
I am not sure I understand the scenario correctly. You lock the datastore by NETCONF `<lock>` and then try to lock it also directly from an application and are surprised you fail to get the second lock? It works exactly as expected, NETCONF locks are translated directly to `sr_lock()` API calls.

**Comment by vshaker** [comment]
Scenario is correct locked the datastore by NETCONF <lock> and then try to lock it also directly from an application
But How do we do it gracefully is the question without application throwing error and coming out of the operation . 

For example , if netconf has locked the DS , Instead of error when i try to lock from application , i want the application to know DS is locked and wait for the lock to realase and how long can the application wait for the lock?

**Comment by michalvasko** [comment]
Right, you can only check for error `SR_ERR_LOCKED` and retry after some reasonable sleep. I will add a timeout for `sr_lock()` in next *sysrepo* version.

**Comment by vshaker** [comment]
Thank you for the response
